platform/qt/src/qt_geojson.cpp

```diff
@@ -132,8 +130,7 @@ mbgl::FeatureIdentifier asMapboxGLFeatureIdentifier(const QVariant &id) {
 mbgl::Feature asMapboxGLFeature(const QMapbox::Feature &feature) {
     mbgl::PropertyMap properties;
     properties.reserve(feature.properties.size());
-    auto it = feature.properties.constBegin();
-    while (it != feature.properties.constEnd()) {
+    for (auto it = feature.properties.constBegin(); it != feature.properties.constEnd(); ++it) {
         properties.emplace(std::make_pair(it.key().toStdString(), asMapboxGLPropertyValue(it.value())));
```

Comment:
Maybe simpler 'for (const auto &pair: map)'?

Reply:
Tried that, but it seems the range-based for loop drops the keys out of the iterator, causing it to just contain values.